### ISSUE TITLE
deken think "sources" is an architecture

### ISSUE THREAD

**Issue by ch-nry** [pure-data/deken#302]
When searching for a package, deken did not show the sources package, if the "hide other architecture" toggle is set (default).
It also remove the binary when installing the source of the object.
I expect deken to understand that a source package can be installed on all architecture, without removing the binary. 
(tested with simplex~ external, deken 0.9.18, pd 0.55)

**Comment by umlaeute** [comment]
> When searching for a package, deken did not show the sources package, if the "hide other architecture" toggle is set (default).

i think this is the correct and expected behaviour.
the package manager targets ready-to-use packages, rather than source distributions (most people will not be able to actually use the source package)

>  It also remove the binary when installing the source of the object.

this one is tricky: what *is* the correct behaviour?
- i'm pretty sure that just not-removing the binaries is the wrong approach: often a source package has a completely different layout than the binary packages (e.g. help-patches are in different folders), leading to a huge mess if we just merge the two
- providing a separate installation directory for source packages might be an option (but adds another confusing configuration)
- and what to do about mixed packages (that contain both binaries and sources)? (obviously they cannot be handled the same as only-source packages)

**Comment by umlaeute** [comment]
one possibility would be to check if the the to-be-installed package is a foreign-architecture (that is: does *not* contain the host architecture), and present an extra dialog, asking to to install to some other place.

**Comment by ch-nry** [comment]
> one possibility would be to check if the the to-be-installed package is a foreign-architecture (that is: does _not_ contain the host architecture), and present an extra dialog, asking to to install to some other place.

what about asking what to do if the lib is already installed I.E the directory is already there (no need to check the architecture) : 
-remove old dir
-merge
-install elsewhere


But the source of the problem may comes that the "source" package is considered as an architecture. And I don't see a good solution to change that.

**Comment by ch-nry** [comment]
> i'd rather not have a popup everytime i'm trying to install an already-installed library.
This should not be very often, so  don't think this popup will be problematic
>but then, nobody will find (or understand) this option, so it's probably not very useful.
agree.

**Comment by umlaeute** [comment]
> This should not be very often, so don't think this popup will be problematic

hmm, dunno.
i guess for libraries that are often updated (e.g. `ELSE`), people will re-install regularly.

**Comment by ch-nry** [comment]
ah! there is a difference between upgrading the lib and installing the same version, but for an other architecture. Can deken know the difference?

**Comment by umlaeute** [comment]
not really. (it has no knowledge about the currently installed version).

in any case, i've now added a dialog if you are uninstalling for an incompatible architecture.
